pkgs/applications/networking/browsers/ladybird/default.nix

```diff
@@ -1,71 +1,117 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchurl
+, fetchzip
 , cmake
 , ninja
-, unzip
 , wrapQtAppsHook
 , libxcrypt
 , qtbase
+, qtmultimedia
 , nixosTests
 }:
 
+let
+  cldr_version = "43.1.0";
+  ucd_version = "15.0.0";
+  emoji_txt_version = "15.0";
+  tzdb_version = "2023c";
+  cacert_version = "2023-05-30";
+  cldr = fetchzip {
+    url = "https://github.com/unicode-org/cldr-json/releases/download/${cldr_version}/cldr-${cldr_version}-json-modern.zip";
+    stripRoot = false;
+    hash = "sha256-+PCYYUWwiDyTfDvSxBAbVRpA7YkuQGKr3ZPlnc9PnKk=";
+  };
+  ucd = fetchzip {
+    url = "https://www.unicode.org/Public/${ucd_version}/ucd/UCD.zip";
+    stripRoot = false;
+    hash = "sha256-jj6bX46VcnH7vpc9GwM9gArG+hSPbOGL6E4SaVd0s60=";
+  };
+  emoji_txt = fetchurl {
+    url = "https://www.unicode.org/Public/emoji/${emoji_txt_version}/emoji-test.txt";
+    hash = "sha256-hEXyOsg4jglr4Z0CYuFPzv+Fb/Ugk/I1bciUhfGoU9s=";
+  };
+  tzdb = fetchzip {
+    url = "https://data.iana.org/time-zones/releases/tzdata${tzdb_version}.tar.gz";
+    stripRoot = false;
+    hash = "sha256-G/IC0EHQcuXUTr2G/6Kn1giC0d1WC9hHIE2fQI26rlM=";
+  };
+  cacert = fetchurl {
+    url = "https://curl.se/ca/cacert-${cacert_version}.pem";
+    hash = "sha256-X63K6QqkrgQRUPji0mw32YBSLNtJ+SP8HhteuNdOca0=";
+  };
+in
 stdenv.mkDerivation (finalAttrs: {
   pname = "ladybird";
-  version = "unstable-2023-01-17";
+  version = "unstable-2023-08-10";
 
   src = fetchFromGitHub {
     owner = "SerenityOS";
     repo = "serenity";
-    rev = "45e85d20b64862df119f643f24e2d500c76c58f3";
-    hash = "sha256-n2mLg9wNfdMGsJuGj+ukjto9qYjGOIz4cZjgvMGQUrY=";
+    rev = "941a9846a34173e1998b623c2664f0cb513fc4f5";
+    hash = "sha256-CckOwQutD5N8ClzOUeyHeyxGv7TK+Wo1xVHcu6La0NY=";
   };
 
   sourceRoot = "${finalAttrs.src.name}/Ladybird";
 
   postPatch = ''
     substituteInPlace CMakeLists.txt \
       --replace "MACOSX_BUNDLE TRUE" "MACOSX_BUNDLE FALSE"
-    # https://github.com/SerenityOS/serenity/issues/17062
-    substituteInPlace main.cpp \
-      --replace "./SQLServer/SQLServer" "$out/bin/SQLServer"
-    # https://github.com/SerenityOS/serenity/issues/10055
-    substituteInPlace ../Meta/Lagom/CMakeLists.txt \
-      --replace "@rpath" "$out/lib"
+
+    mkdir -p Caches
+    cp -r ${cldr} Caches/CLDR
+    cp -r ${ucd} Caches/UCD
+    cp -r ${tzdb} Caches/TZDB
+    mkdir Caches/CACERT
+    cp -r ${cacert} Caches/CACERT/cacert-${cacert_version}.pem
+
+    chmod +w Caches/UCD
+    chmod +w Caches/CLDR
+    chmod +w Caches/TZDB
+    chmod +w Caches/CACERT
+
+    cp ${emoji_txt} Caches/UCD/emoji-test.txt
+
+    echo -n ${cldr_version} > Caches/CLDR/version.txt
+    echo -n ${ucd_version} > Caches/UCD/version.txt
+    echo -n ${tzdb_version} > Caches/TZDB/version.txt
+    echo -n ${cacert_version} > Caches/CACERT/version.txt
+
+    chmod -w Caches/UCD
+    chmod -w Caches/CLDR
+    chmod -w Caches/TZDB
+    chmod -w Caches/CACERT
```

Comment:
This can be done with `postFetch` in `fetchzip` (unless we end up doing separate packages as was suggested in the other thread)

Reply:
I did this for the CLDR json. I think the current setup with preConfigure looks "slightly" better than the way I had it before, but still not super great. Since I pulled the rest of the files from existing packages it still looks a bit gnarly.